### PR TITLE
Save donation data to avoid duplicated donation notifications

### PR DESCRIPTION
**Pull request by denys-medynskyi**
### Description

To avoid duplicated donations let's save donation and avoid duplicate wtih unique index on `order_reference` column

### How to test instructions

- rails s
- Run in terminal
 ```curl --request POST 'http://localhost:3000/donate/webhook' \
--header 'Content-Length:593' \
--header 'X-Forwarded-Proto:https' \
--header 'X-Forwarded-For:88.85.72.203' \
--header 'Connection:close' \
--header 'Accept:*/*' \
--header 'X-Request-Start:t=1575791662.136' \
--header 'Host:pivorak.com' \
--header 'Version:HTTP/1.0' \
--header 'Content-Type:application/x-www-form-urlencoded' \
--header 'X-Forwarded-Ssl:on' \
--data '{"merchantAccount":"pivorak_com","orderReference":"WFP-2102129-5decabc01799e","merchantSignature":"8c3a5a8211b9556d475be7a1f5f5b5d3","amount":1,"currency":"UAH","authCode":"855512","email":"denys.medynskyi@gmail.com","phone":"380637589464","createdDate":1575791552,"processingDate":1575791590,"cardPan":"53****6819","cardType":"MasterCard","issuerBankCountry":"Ukraine","issuerBankName":"MONObank","recToken":"","transactionStatus":"Approved","reason":"Ok","reasonCode":1100,"fee":0.13,"paymentSystem":"card","acquirerBankName":"WayForPay","cardProduct":"debit","clientName":"DENYS MEDYNSKYI"}=None'
```
- You should see notification in #test-notifications channel
- Run in terminal again:
```
curl --request POST 'http://localhost:3000/donate/webhook' \
--header 'Content-Length:593' \
--header 'X-Forwarded-Proto:https' \
--header 'X-Forwarded-For:88.85.72.203' \
--header 'Connection:close' \
--header 'Accept:*/*' \
--header 'X-Request-Start:t=1575791662.136' \
--header 'Host:pivorak.com' \
--header 'Version:HTTP/1.0' \
--header 'Content-Type:application/x-www-form-urlencoded' \
--header 'X-Forwarded-Ssl:on' \
--data '{"merchantAccount":"pivorak_com","orderReference":"WFP-2102129-5decabc01799e","merchantSignature":"8c3a5a8211b9556d475be7a1f5f5b5d3","amount":1,"currency":"UAH","authCode":"855512","email":"denys.medynskyi@gmail.com","phone":"380637589464","createdDate":1575791552,"processingDate":1575791590,"cardPan":"53****6819","cardType":"MasterCard","issuerBankCountry":"Ukraine","issuerBankName":"MONObank","recToken":"","transactionStatus":"Approved","reason":"Ok","reasonCode":1100,"fee":0.13,"paymentSystem":"card","acquirerBankName":"WayForPay","cardProduct":"debit","clientName":"DENYS MEDYNSKYI"}=None'
```
- You should not see notification in #test-notifications channel

### CODE REVIEW

File: db/migrate/20200330085332_create_donation_data.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class CreateDonationData < ActiveRecord::Migration[5.2]
+  def change
+    create_table :donation_data do |t|
+      t.string :order_reference
+      t.timestamps null: false
```
Comment (skrix) [inline db/migrate/20200330085332_create_donation_data.rb]: `null: false` is by default

File: spec/services/donate/send_new_donate_notification_spec.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # frozen_string_literal: true
 
 describe Donate::SendNewDonateNotification do
-  let(:call) { described_class.call(amount: amount, currency: 'UAH') }
+  let(:call) { described_class.call(amount: amount, currency: 'UAH', orderReference: order_reference) }
   let(:amount) { 5 }
+  let(:order_reference) { "WFP-2102129-5decabc01799e" }
```
Comment (skrix) [inline spec/services/donate/send_new_donate_notification_spec.rb]: single quotes

Comment (skrix) [inline spec/services/donate/send_new_donate_notification_spec.rb]: alignment of `{}`

File: db/schema.rb [hunk 3]
```diff
@@ -130,6 +130,13 @@
     t.text "notes"
   end
 
+  create_table "donation_data", force: :cascade do |t|
```
Comment (skrix) [inline db/schema.rb:133]: donations will be dropped and this table will be renamed, or still the same?

File: app/services/donate/send_new_donate_notification.rb [hunk 4]
```diff
@@ -3,15 +3,19 @@
 # Donate::SendNewDonateNotification.call('amount' => 5, 'currency' => 'USD')
```
Comment (skrix) [inline app/services/donate/send_new_donate_notification.rb]: add `order_reference` explanation than

**Review by skrix — CHANGES_REQUESTED** [review]

File: db/migrate/20200330085332_create_donation_data.rb (hunk 1 above)
Comment (denys-medynskyi) [reply db/migrate/20200330085332_create_donation_data.rb]: good

File: spec/services/donate/send_new_donate_notification_spec.rb (hunk 2 above)
Comment (denys-medynskyi) [reply spec/services/donate/send_new_donate_notification_spec.rb]: agree

Comment (denys-medynskyi) [reply spec/services/donate/send_new_donate_notification_spec.rb]: Is it a rubocop rule?

File: db/schema.rb (hunk 3 above)
Comment (denys-medynskyi) [reply db/schema.rb:133]: Probably renamed

**Review by skrix — APPROVED** [review]

**Pull request merged**